tests/director/api/application_api_test.go

```diff
@@ -465,17 +466,77 @@ func TestQuerySpecificApplication(t *testing.T) {
 	err = tc.RunOperation(context.Background(), request, &actualApp)
 	require.NoError(t, err)
 	require.NotEmpty(t, actualApp.ID)
-	createdID := actualApp.ID
-	defer unregisterApplication(t, actualApp.ID)
+	appID := actualApp.ID
+	defer unregisterApplication(t, appID)
 
-	// WHEN
-	queryAppReq := fixApplicationRequest(actualApp.ID)
-	err = tc.RunOperation(context.Background(), queryAppReq, &actualApp)
-	saveExampleInCustomDir(t, queryAppReq.Query(), queryApplicationCategory, "query application")
+	t.Run("Query Application With Consumer User", func(t *testing.T) {
+		actualApp := graphql.Application{}
 
-	//THEN
-	require.NoError(t, err)
-	assert.Equal(t, createdID, actualApp.ID)
+		// WHEN
+		queryAppReq := fixApplicationRequest(appID)
+		err = tc.RunOperation(context.Background(), queryAppReq, &actualApp)
+		saveExampleInCustomDir(t, queryAppReq.Query(), queryApplicationCategory, "query application")
+
+		//THE
+		require.NoError(t, err)
+		assert.Equal(t, appID, actualApp.ID)
+	})
+
+	ctx := context.Background()
+
+	runtime := registerRuntime(t, ctx, "runtime-test")
+	defer unregisterRuntime(t, runtime.ID)
+
+	scenarios := []string{defaultScenario, "test-scenario"}
+
+	// update label definitions
+	updateScenariosLabelDefinitionWithinTenant(t, ctx, testTenants.GetDefaultTenantID(), scenarios)
+	defer updateScenariosLabelDefinitionWithinTenant(t, ctx, testTenants.GetDefaultTenantID(), scenarios[:1])
+
+	runtimeConsumer := tc.NewOperation(ctx).WithConsumer(&jwtbuilder.Consumer{
+		ID:   runtime.ID,
+		Type: jwtbuilder.RuntimeConsumer,
+	})
+
+	t.Run("Query Application With Consumer Runtime in same scenario", func(t *testing.T) {
+		// set application scenarios label
+		setApplicationLabel(t, ctx, appID, scenariosLabel, scenarios[1:])
+		defer setApplicationLabel(t, ctx, appID, scenariosLabel, scenarios[:1])
+
+		// set runtime scenarios label
+		setRuntimeLabel(t, ctx, runtime.ID, scenariosLabel, scenarios[1:])
+		defer setRuntimeLabel(t, ctx, runtime.ID, scenariosLabel, scenarios[:1])
+
+		// query application
```

Comment:
query application comment here above the section of creating an empty app is abit confusing